### PR TITLE
Show a different text in the feed for the new savings rewards

### LINKED ISSUE #655: Show a different text in the feed for the new savings rewards
### What would you like?
For the new cUSD rewards, the title and description match the ones in Figma: https://www.figma.com/file/31HUGhIHLXEm1qwVpHV1QR/CELO-Rewards?node-id=1763%3A1350

### Why is this needed?
We release a new program and we want a new text for new rewards

### PR DESCRIPTION
**Pull request by gnardini**
### Description

Updated the feed for the new rewards program. The title text is now different (`Earned CELO` -> `CELO Network`).

I created a new array in Firebase called `rewardsSenders`.
In the past for CELO rewards we added new entries to the `addressesExtraInfo` node with a boolean indicating that it's a reward, but that one is growing a lot with a lot of identical elements which have the same name & image (the name is being ignored and set client-side). By having this array instead we can leave the `addressesExtraInfo` node for the initial purpose (special addresses like Moola or cash in providers) and not dirty it with more conditionals.

simulator_screenshot_60C61F6A-AF13-4756-959E-D8C25D978997

### Other changes

Converted the recipients reducer to redux toolkit.

### Tested

Set one of my test addresses as a rewards sender to test.

### How others should test

Hard to test this one. 
You can send me your address and I'll make a transaction to you from a fake rewards sender address. Once you have an address. I can also set any address you want as a 'rewards sender address'.
For any address marked as a rewards address, all incoming transactions from it should say `CELO Network` in the title and `Weekly Earnings` in the subtitle.
In the transaction detail for these rewards, the `Learn more` button should redirect to the rewards screen.

### Related issues

- Fixes #655

### Backwards compatibility

N/A

### CODE REVIEW

File: packages/mobile/src/recipients/reducer.ts [hunk 1]
```diff
@@ -12,44 +12,54 @@ export interface State {
...
   // and includes CIP8 profile data if available
   valoraRecipientCache: AddressToRecipient
+  rewardsSenders: string[]
 }
 
-const initialState = {
+const initialState: State = {
   phoneRecipientCache: {},
   valoraRecipientCache: {},
+  rewardsSenders: [],
```
Comment (etuleu) [inline packages/mobile/src/recipients/reducer.ts:21]: just an idea but would it be cleaner to create a new rewards package?

**Review by etuleu — APPROVED** [review]

File: packages/mobile/src/recipients/reducer.ts [hunk 2]
```diff
@@ -12,44 +12,54 @@ export interface State {
   // valoraRecipientCache contains accounts that the user has sent/recieved transactions from,
   // and includes CIP8 profile data if available
   valoraRecipientCache: AddressToRecipient
+  rewardsSenders: string[]
```
Comment (jeanregisser) [inline packages/mobile/src/recipients/reducer.ts:15]: More correct if there's no migration:
```suggestion
  rewardsSenders: string[] | undefined
```

File: packages/mobile/src/recipients/reducer.ts [hunk 3]
```diff
@@ -12,44 +12,54 @@ export interface State {
...
+    }))
+    .addCase(updateValoraRecipientCache, (state, action) => ({
+      ...state,
+      valoraRecipientCache: { ...state.valoraRecipientCache, ...action.payload },
+    }))
+    .addCase(setRewardsSenders, (state, action) => ({
+      ...state,
+      rewardsSenders: action.payload,
+    }))
+})
```
Comment (jeanregisser) [inline packages/mobile/src/recipients/reducer.ts:56]: Nice! I think we should go even further and use `createSlice` instead, but let's do this in separate PRs.
https://redux-toolkit.js.org/api/createslice

File: packages/mobile/src/recipients/reducer.ts [hunk 4]
```diff
@@ -12,44 +12,54 @@ export interface State {
...
-  switch (action.type) {
-    case REHYDRATE: {
+const rehydrate = createAction<any>(REHYDRATE)
+export const setPhoneRecipientCache = createAction<NumberToRecipient>(
+  'RECIPIENTS/SET_PHONE_RECIPIENT_CACHE'
+)
+export const updateValoraRecipientCache = createAction<AddressToRecipient>(
+  'RECIPIENTS/SET_VALORA_RECIPIENT_CACHE'
+)
+export const setRewardsSenders = createAction<string[]>('RECIPIENTS/SET_REWARDS_SENDERS')
```
Comment (jeanregisser) [inline packages/mobile/src/recipients/reducer.ts:31]: We should avoid using setter actions and [model them as events](https://redux.js.org/style-guide/style-guide#model-actions-as-events-not-setters).

**Review by jeanregisser — COMMENTED** [review]
This works for now, but I think we should think about this problem of displaying specific TXs differently a bit more globally.

Initially the intent was to add new transaction types to `blockchain-api`
See https://github.com/celo-org/wallet/blob/1c0760de7edd67899919b253ff50678e6bd5fa4a/packages/blockchain-api/src/schema.ts#L147-L159

Of course it still requires changing the client to support the new type.
But we could get more flexibility by passing a feed item title and icon too.

**Comment by gnardini** [comment]
I agree on returning stuff from the blockchain-api. I think we should return as much info as we can from there. I even created a PR with a first step a while back: https://github.com/celo-org/wallet/pull/427

The reason I closed that PR and added this client-side only is for short-term simplicity since I think we should make bigger changes to the blockchain-api so that it returns the item feed from its own database instead of querying blockscout. Might make sense to change this before then though.

File: packages/mobile/src/recipients/reducer.ts (hunk 2 above)
Comment (gnardini) [reply packages/mobile/src/recipients/reducer.ts:15]: I added a spread of the `initialState` to the `rehydrate` action so that this isn't necessary, otherwise we'd need to add null checks in different places.

File: packages/mobile/src/recipients/reducer.ts [hunk 5]
```diff
@@ -12,44 +12,55 @@ export interface State {
...
+)
+export const rewardsSendersFetched = createAction<string[]>('RECIPIENTS/REWARDS_SENDERS_FETCHED')
+
+export const recipientsReducer = createReducer(initialState, (builder) => {
+  builder
+    .addCase(rehydrate, (state, action) => {
+      // hack to allow rehydrate actions here
+      const hydrated = getRehydratePayload((action as unknown) as RehydrateAction, 'recipients')
       return {
+        ...initialState,
```
Comment (jeanregisser) [inline packages/mobile/src/recipients/reducer.ts:39]: I think `initialState` is not even needed, since `state` should be the `initialState` already.

**Review by jeanregisser — APPROVED** [review]
LGTM 👍

**Comment by MuckT** [comment]
https://valora.testquality.com requirement [R45](https://valora.testquality.com/project/12128/testcase/292996/requirement/11539).
